src/librustc_error_codes/error_codes/E0657.md

```diff
@@ -0,0 +1,64 @@
+The `impl Trait` can only capture lifetimes bound at the fn or impl level.
+
+Erroneous code example:
+
+```compile_fail,E0657
+#![allow(warnings)]
```

Comment:
Why?

Reply:
@GuillaumeGomez  Because the code will give warnings for not using the function, trait, etc in the main function.

Reply:
You don't see them when reading the index error so it's not useful in here.

Reply:
sure.